### DISCUSSION TITLE
How to convert XYZ atom coordinate data in double precision (16 decimal places) to create TRR for use with Gromacs?

### DISCUSSION THREAD

**Discussion by paul31** [MDAnalysis/mdanalysis#4736] [Q&A] [answered]
Hello there!

I'm starting to use MDAnalysis and so far it's working very well, thanks to all the people who have contributed to this project!

My question is about outputting atom coordinates in double precision. I have atom XYZ coordinates in double precision (16 decimal places) and would like to convert them to TRR files with full precision for use in Gromacs simulations.

I hope someone can help, thanks for reading!

Paul

**Comment by hmacdope** [comment, marked as answer]
Hi @paul31, MDAnalysis doesn't currently support full `double` precision for timesteps or in the low level interface in [libmdaxdr](url), instead requiring 32 bit float precision only however this is something we are planning on working on for MDAnalysis 3.0 . See issue #3703,  #3874 for discussion. 

While we do acknowledge the need to improve our support for different precision, my question to you would be what exactly do you need all that precision for? It is many times unnecessary to compute things in full precision, differing on a case by case basis. 

If you really want to do this I would say you would probably need to write a program in C to read them, e.g with this C file and corresponding header. https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/lib/formats/src/xdrfile_trr.c, but may be a challenge.

**Reply by paul31** [reply]
Hello @hmacdope thanks for your answer, that's really helpful to know!

So my aim is to couple together Gromacs with a Monte Carlo code to simulate rigid molecules. The MC code is in double precision so any loss of accuracy leads to incorrect bond lengths and angles.

I can get Gromacs to output atom coordinates in double precision, so the MD->MC part is fine.

But going the other way, MC->MD, requires me to get Gromacs to read double precision atom coordinates somehow. I was hoping MDAanalysis could be used to create input TRR files in double precision.

It's good to hear that this is something that might be worked on in the future and I will look into the file you linked to.

Thanks again for the helpful reply it will save me a lot of time!
